lib/entities/plans/index.ts

```diff
@@ -0,0 +1,30 @@
+/** @format **/
+
+import { buildApiUrl, NetworkClient } from "../../network";
+import { wrapWithErrorHandler } from "../../network/utils";
+import { IApplications } from "./models";
+
+export class Plans {
+  networkClient: NetworkClient;
+  baseUrl: string = "applications";
+
+  constructor(networkClient: NetworkClient) {
+    this.networkClient = networkClient;
+  }
+
+  /**
+   * Retrieve all applications that matches the specified plan identifier
+   * @param _planIdentifier - Identifier of the plan that the application belongs to.
+   * @returns All applications on the plan including its relevent information about the app
+   */
+
+  public async getAllApplications(_planIdentifier:string): Promise<IApplications[] | undefined> {
+    const urlSegments = ["plans", _planIdentifier, this.baseUrl];
+
+    return await wrapWithErrorHandler(async () => {
+      const url = buildApiUrl(urlSegments);
+      const res = await this.networkClient.get<IApplications[]>(url);
+      return res;
+    });
+  }
```

Comment:
I think this should be within the `Applications` class and then we can rename the method to just `getAll` instead. As `Plan` is the root parent entity, given that all other entities can be parented to a Plan, I don't think we will have a `Plan` class within the SDK as it has the potential to get bloated over time.

Reply:
The asociated models and utils can be moved to the `entities/applications` folder.

Reply:
That's good to know! I was actually tossing between adding in a new plan folder or just including it within the applications folder but as the endpoint begun with `/plan/` I thought it would be best to match with that instead. but that makes sense keep app-related endpoints within the applications area to not create a plan area